### DISCUSSION TITLE
how to deploy project to the sub path?

### DISCUSSION THREAD

**Discussion by yangshangbo** [roxiness/routify#406] [Q&A] [answered]
i deployed my project to http://106.14.73.216/svelte-routify/ path;
when i visit my project, the vendorjs catch an error like that below
` vendor.e29876a0.js:1 Uncaught (in promise) Error: router: "[default]" could not find route: /svelte-routify `

**Comment by jakobrosenberg** [comment]
Is this Routify 3 and do you have a path by that name?

**Reply by yangshangbo** [reply]
Yes, It's Routify 3.
What means of that `have a path by that name`?

**Reply by jakobrosenberg** [reply]
My mistake. I overlooked the title about "sub path", so I wasn't sure if you had a page in your project called `svelte-routify`.

I'll try get back to you with an example on how to do this. 🙂

**Reply by ghostdevv** [reply]
I found this example from the discord:

```html
<script>
  import { Router } from '@roxi/routify'

  const config = {
    urlTransform: {
      apply: url => `/my-base${url}`, //external URL
      remove: url => url.replace('/my-base', ''), //internal URL
    },
  }
</script>

<Router {config} />
```

replace my-base with your base url

**Reply by jakobrosenberg** [reply]
I think that's for R2.

**Reply by ghostdevv** [reply]
my bad

**Comment by jakobrosenberg** [comment, marked as answer]
Sorry about the delay on this. The last few days were crazy. 😅
I've created an example [here](https://github.com/roxiness/routify/blob/next/examples/subpath/src/App.svelte)


<details>

<summary>
Alternatively you can also create the router in the script
</summary>

```html
<script>
    import { Router, createRouter } from '@roxi/routify'
    import routes from '../.routify/routes.default.js'

    const router = createRouter({
        routes,
        urlRewrite: {
            toExternal: url => '/subpath' + url,
            toInternal: url => url.replace(/^\/subpath/, ''),
        },
    })
</script>

<Router {router} />
```

</details>

**Reply by yangshangbo** [reply]
It works, thanks for your help
